core/src/calleeFunctions/UniswapV3Callee.ts

```diff
@@ -40,23 +39,28 @@ const getMarketPrice = async function (
     collateral: CollateralConfig,
     marketId: string,
     collateralAmount: BigNumber
-): Promise<BigNumber> {
-    // convert collateral into DAI
-    const { route, fees } = await getRouteAndGasQuote(network, collateral.symbol, collateralAmount, marketId);
-    if (!route) {
-        throw new Error(`No route found for ${collateral.symbol} to DAI`);
-    }
-    const pools = await routeToPool(network, route, collateral.symbol, fees);
-    const daiAmount = await convertCollateralToDaiUsingPool(
+): Promise<{ price: BigNumber; pools: Pool[] }> {
+    const { route, fees, totalPrice, pools } = await getRouteAndGasQuote(
```

Comment:
Nit: I would remove `getRouteAndGasQuote` whatsoever and move its logic here, inside `getMarketPrice`. Because it's just adds a very strange layer of abstraction which doesn't add any value, but complicates understanding of what's going on